lib/Driver/ToolChains.cpp

```diff
@@ -318,6 +314,23 @@ ToolChain::constructInvocation(const CompileJobAction &job,
   addCommonFrontendArgs(*this, context.OI, context.Output, context.Args,
                         Arguments);
 
+  // Pass along an -import-objc-header arg, replacing the argument with the name
+  // of any input PCH to the current action if one is present.
+  if (context.Args.hasArgNoClaim(options::OPT_import_objc_header)) {
+    bool ForwardAsIs = true;
+    for (auto *IJ : context.Inputs) {
+      if (!IJ->getOutput().getAnyOutputForType(types::TY_PCH).empty()) {
+        Arguments.push_back("-import-objc-header");
+        addInputsOfType(Arguments, context.Inputs, types::TY_PCH);
```

Comment:
It does bug me a little that we end up searching the input list *twice* here, but I guess it will usually be short. You could also have `addInputsOfType` return a boolean or a count, though (and explicitly get the value out of the argument in the non-PCH case, rather than forwarding with `AddLastArg`).